dateutil/parser.py

```diff
@@ -725,11 +726,20 @@ def _parse(self, timestr, dayfirst=None, yearfirst=None, fuzzy=False,
                     # Token is a number
                     len_li = len(l[i])
 
-                    if (len(ymd) == 3 and len_li in (2, 4) and
-                        res.hour is None and
-                        (i + 1 >= len_l or
-                         (l[i + 1] != ':' and
-                          info.hms(l[i + 1]) is None))):
+                    hms_idx = _find_hms_idx(i, l, info, allow_jump=True)
+                    if hms_idx is not None:
```

Comment:
Yeah, I'm fine with calling it twice.

Honestly, I think if we implement #125 and add format caching, we'll see some dramatic speedup in nearly all real-world use-cases, so little stuff won't really "add up" in any real sense.